pkg/prometheus/promcfg.go

```diff
@@ -535,6 +535,14 @@ func mergeAttachMetadataWithScrapeClass(attachMetadata *monitoringv1.AttachMetad
 	}
 }
 
+func mergeFallbackScrapeProtocolWithScrapeClass(fallbackScrapeProtocol *monitoringv1.ScrapeProtocol, scrapeClass monitoringv1.ScrapeClass) *monitoringv1.ScrapeProtocol {
+	if fallbackScrapeProtocol == nil {
```

Comment:
I think we can do something like that.
```suggestion
	return ptr.Deref(fallbackScrapeProtocol, scrapeClass.FallbackScrapeProtocol)
```